src/modules/submissions.js

```diff
@@ -459,7 +459,26 @@
 			user = this.params.u;
 
 		if ( user ) {
-			deferred.resolve( user );
+			AFCH.api.get( {
+				action: 'query',
+				format: 'json',
+				list: 'logevents',
+				formatversion: '2',
+				letype: 'renameuser',
+				letitle: ( new mw.Title( user, 2 ) ).getPrefixedText()
+			} ).done( function ( resp ) {
+				var logevents = resp.query.logevents;
+
+				if ( logevents.length ) {
+					var newName = logevents[ 0 ].params.newuser;
+					this.params.u = newName;
+					this.getSubmitter().done( function ( user ) {
+						deferred.resolve( user );
+					} );
```

Comment:
What if it fails? The deferred is not being rejected in that case which leaves it hanging in a never-resolved/never-rejected state. 
It would be good to convert the method to return the deferreds/promises rather than manually resolve/reject in a then block. Fine to leave for later as it's problem common to the entire codebase.